mlrun/artifacts/manager.py

```diff
@@ -191,6 +191,8 @@ def log_artifact(
         if db_key is None:
             # set the default artifact db key
             if producer.kind == "run":
+                # In case of a run kind producer, we enable a key that is distinct from the db_key,
+                # in order to allow the user to obtain all the runs associated with a result.
```

Comment:
`all the runs associated with a result` or `all results associated with a run`? I mean, what is the exceptional narrative here? user wants to see all runs yield a specific artifact or all run produced artifacts?

Reply:
all run's associated with the result,  so that the user receives all runs and can determine which one is of interest to him. This means that when a user queries an artifact, he will obtain results with numerous db_keys (which can differ from the key in his query), but this is purposeful because all of the runs are relevant.
In the scenario we thought was a bug, we attempted to edit an existing artifact by providing a key in the request and another key in the artifact's body, which resulted in the creation of a new artifact, and then when querying with the request key, we received multiple results, which appears to be an unexpected behavior, but in this case it is intentional.

Reply:
I'm not sure I understand your comment with `runs associated with a result`, where is the behavior you are referring to in the code?